src/boot/bootctl.c

```diff
@@ -1693,13 +1693,21 @@ static int verb_set_default(int argc, char *argv[], void *userdata) {
                 if (r < 0 && r != -ENOENT)
                         return log_error_errno(r, "Failed to remove EFI variale: %m");
         } else {
-                _cleanup_free_ char16_t *encoded = NULL;
+                _cleanup_free_ char16_t *target = NULL;
+                size_t target_size = 0;
 
-                encoded = utf8_to_utf16(argv[1], strlen(argv[1]));
-                if (!encoded)
-                        return log_oom();
+                if (streq(argv[1], "@current")) {
+                        r = efi_get_variable(EFI_VENDOR_LOADER, "LoaderEntrySelected", NULL, (void *) &target, &target_size);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to get EFI variable: %m");
```

Comment:
error should really say which one we are talking about I think, i.e. mention `LoaderEntrySelected` in the log message, please

Reply:
(i mean, as long as this operation only dealt with one EFI variable, the one this verb is supposed to say I think it's fine to omit which variable we are talking about, but now that we deal with multiple we should clarify which one we have issues with. Maybe hence update the log message for the efi_set_variable() call below too, slightly)

Reply:
oh, I kind of assumed the `%m` would report that. otherwise I thought the same.